ui-admin/src/study/StudySettings.tsx

```diff
@@ -46,9 +48,33 @@ export function StudyEnvConfigView({ studyEnvContext, portalContext }:
                                        {studyEnvContext: StudyEnvContextT, portalContext: LoadedPortalContextT}) {
   const [config, setConfig] = useState(studyEnvContext.currentEnv.studyEnvironmentConfig)
   const [isLoading, setIsLoading] = useState(false)
+  const [kitTypeOptions, setKitTypeOptions] = useState<{ value: string, label: string }[]>([])
+  const [selectedKitTypes, setSelectedKitTypes] = useState<{ value: string, label: string }[]>([])
 
   const { user } = useUser()
 
+  const studyEnvParams = {
+    portalShortcode: portalContext.portal.shortcode,
+    studyShortcode: studyEnvContext.study.shortcode,
+    envName: studyEnvContext.currentEnv.environmentName as EnvironmentName
+  }
+
+  const loadAllowedKitTypes = async () => {
+    const allowedKitTypes = await Api.fetchAllowedKitTypes(studyEnvParams)
+    const allowedKitTypeOptions = allowedKitTypes.map(kt => ({ value: kt.name, label: kt.displayName }))
+    setKitTypeOptions(allowedKitTypeOptions)
+  }
+
+  const loadConfiguredKitTypes = async () => {
+    const selectedKitTypes = await Api.fetchKitTypes(studyEnvParams)
+    setSelectedKitTypes(selectedKitTypes.map(kt => ({ value: kt.name, label: kt.displayName })))
+  }
+
+  useEffect(() => {
+    loadAllowedKitTypes()
+    loadConfiguredKitTypes()
+  }, [studyEnvContext])
```

Comment:
```suggestion
  }, [studyEnvContext.curretEnvPath])
```
making the dependency the path (string) rather than the object reference  is more concrete